dangerzone/gui/main_window.py

```diff
@@ -276,6 +279,15 @@ def __init__(self, dangerzone: DangerzoneGui) -> None:
         self.setLayout(layout)
 
     def documents_selected(self, new_docs: List[Document]) -> None:
+        # Get previously selected documents
+        previous_docs = self.dangerzone.get_unconverted_documents()
+        docs_removed = []
+        for prev_doc in previous_docs:
+            if prev_doc not in new_docs:
+                docs_removed.append(prev_doc)
+        for doc in docs_removed:
+            self.dangerzone.remove_document(doc)
```

Comment:
General question, is the removal logic actually needed? I'm asking because I'd assume that clicking the "Change Selection" button would simply remove all the previously selected documents, so there would be no need for removing duplicates, `Document` instance comparison, etc.

Reply:
The issue is that the UI doesn't "react" when things change. So we need to register the removal so that the documents list widget gets the info that certain objects need to removed. Does this make sense?

Reply:
Sure I get it, but my question is why can't we just nuke the previously selected documents. For example:

```suggestion
        self.dangerzone.documents = []
        self.documents_list.docs_list = []
        self.documents_list.docs_list_widget_map = {}
        self.documents_list.clear()
```

Reply:
I have now implemented an iteration of this in https://github.com/freedomofpress/dangerzone/pull/439/commits/0916344c2ed51bb5f7f7dc7fbd7bf2d38f57e607. The diff looks a bit more complex because I changed the order in `ContentWidget.documents_selected`, but it is essentially the same.

I implemented the following in `documents_list` itself, overriding the clear() method
```
        self.documents_list.docs_list = []
        self.documents_list.docs_list_widget_map = {}
```

Reply:
Perfect, thanks a lot for this. I think we can also remove the code for checking document equality as well, but I'll leave it up to you.